extension/lib/logger.js

```diff
@@ -108,27 +104,43 @@ Logger.prototype._writeAsync = function Logger_writeAsync(aMessage, aCallback) {
   // For testing purposes send the message to stdout
   //dump(aMessage + '\n');
 
-  // Create an output stream to write to file
-  var foStream = Cc["@mozilla.org/network/file-output-stream;1"]
-                 .createInstance(Ci.nsIFileOutputStream);
-  foStream.init(this._file, 0x02 | 0x08 | 0x10, PERMS_FILE, foStream.DEFER_OPEN);
-
-  // Write asynchronously to buffer;
-  // Input and output streams are closed after write
-  var iStream = this._converter.convertToInputStream(aMessage);
-  NetUtil.asyncCopy(iStream, foStream, function (status) {
-    if (!components.isSuccessCode(status)) {
-      var errorMessage = new Error("Error while writing to file: " + status);
-      console.error(errorMessage);
+  function callback() {
+    if (typeof(aCallback) === "function") {
+      try {
+        aCallback();
+      } catch (e) {
+        Cu.reportError(e);
+      }
```

Comment:
> I don't see why we have to wrap a passed-in callback into another callback. That only complicates the code.

Because it is used twice.

> Hm, so we have a synchronous method now for writing log content?

Yes, this PR is like a refactor: `_writeAsync` now able to return a promise, but that piece isn't done yet. So it do some magic with callbacks instead.

To be complete for this PR rewrite is needed: migrate `Logger`'s callback API to a promise API and use that in the tests.